### ISSUE TITLE
Incorporate K8s best practices guide into K8s docs

### ISSUE THREAD

**Issue by bentsherman** [nextflow-io/nextflow#2773]
I wrote a [K8s best practices guide](https://github.com/seqeralabs/nf-k8s-best-practices) a while ago but I think it would be better to incorporate this guide into the Nextflow Kubernetes docs. The Nextflow docs already has some information about how to prepare a K8s cluster, troubleshooting, etc, so the best practices are mostly just expansions on those sections.

Some specific items:
- when to use Kubernetes
- how to setup Nextflow K8s permissions ("Cluster preparation" in Tower user docs)
- Caveats about kuberun, how to deploy your own workflow pod
- Diagnosing errors, pod failures

I can work on this once the general docs improvements PR (#2698) is merged as that PR makes some changes to the K8s page.

**Comment by bentsherman** [comment]
I decided to archive the best practices guide. Most of the content is either no longer relevant or not really fitting for the reference docs, but they are being covered by various blog posts.
